core/src/main/java/org/elasticsearch/bootstrap/Security.java

```diff
@@ -212,7 +226,36 @@ static Permissions createPermissions(Environment environment) throws IOException
             // we just need permission to remove the file if its elsewhere.
             policy.add(new FilePermission(environment.pidFile().toString(), "delete"));
         }
-        return policy;
+    }
+    
+    static void addBindPermissions(Permissions policy, Settings settings) throws IOException {
+        // http is simple
+        String httpRange = settings.get("http.netty.port", 
+                               settings.get("http.port", 
+                                       NettyHttpServerTransport.DEFAULT_PORT_RANGE));
+        policy.add(new SocketPermission("localhost:" + httpRange, "listen,resolve"));
+        // transport is waaaay overengineered
+        Map<String, Settings> profiles = settings.getGroups("transport.profiles", true);
```

Comment:
The code in NettyTransport does much more with it's settings. See for example NettyTransport.createFallbackSettings and how it's used . I agree what it does is probably an overkill, but can we make the methods there static and use the results setting here (using static strings for settings)?